modules/canopennode/CO_driver_target.h

```diff
@@ -39,12 +39,10 @@ extern "C" {
 #define CO_USE_LEDS 1
 #endif
 
-#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#ifndef CONFIG_BIG_ENDIAN
 #define CO_LITTLE_ENDIAN
-#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
-#define CO_BIG_ENDIAN
 #else
-#error "Unsupported endianness"
+#define CO_BIG_ENDIAN
 #endif
```

Comment:
Take for instance this change. If `CONFIG_BIG_ENDIAN` is not defined, we define `CO_LITTLE_ENDIAN`. Otherwise we define `CO_BIG_ENDIAN`. This would read much naturally as:

```
#ifdef CONFIG_BIG_ENDIAN
#define CO_BIG_ENDIAN
#else
#define CO_LITTLE_ENDIAN
#endif
```

The same logic - in my mind - applies to all the other `#ifndef ... #else ... #endif` statements, but the above is a very literal example.

Reply:
Maybe we could go case by case then. Since this is mostly about personal preference, I can change this particular one to match what you describe, and leave the other as `#if !defined()` unless a particular maintainer asks me to change it.